### PR TITLE
[Feature][Model] Use common FusedMoE for LingBot Video

### PR DESCRIPTION
**Pull request by wtz2333**
## Purpose

LingBot Video currently executes routed experts through a model-specific
`torch._grouped_mm` implementation. This duplicates expert execution logic already
provided by vLLM and prevents the model from using the shared fused MoE backend.

This PR integrates LingBot Video with vLLM's common `FusedMoE` runner while preserving
its sigmoid routing, grouped top-k selection, correction bias, routed scaling, and
padding behavior. It also moves transformer weight loading to the shared diffusion
loader so checkpoint tensors can be mapped into the fused expert layout and undergo
the backend's post-load processing. Text encoder and VAE loading remain unchanged.

The goal is to reduce MoE execution latency and model-specific maintenance without
changing the public generation API or checkpoint format. Numerical and end-to-end
video comparisons are reported below because fused execution can change floating-point
rounding and does not guarantee bitwise-identical outputs.

## What Changes

| File | Change |
|---|---|
| `vllm_omni/diffusion/models/lingbot_video/lingbot_video_transformer.py` | Replace custom routed-expert execution with common `FusedMoE`; preserve routing and padding semantics; supply the vLLM forward context; map `w1`/`w3` into `w13_weight` and router parameters into the gate layout while retaining FP32-sensitive parameters. |
| `vllm_omni/diffusion/models/lingbot_video/pipeline_lingbot_video.py` | Declare the transformer weight source for `DiffusersPipelineLoader` and delegate loading to `AutoWeightsLoader`, replacing eager transformer `from_pretrained` loading. |
| `tests/diffusion/models/lingbot_video/test_lingbot_video_transformer.py` | Cover common-runner configuration, padding compaction/restoration, checkpoint mapping, FP32 parameter preservation, and the compile boundary. |
| `tests/diffusion/models/lingbot_video/test_pipeline_lingbot_video.py` | Test transformer weight-source registration and loader delegation; remove assertions for the obsolete loading contract. |
| `tests/diffusion/models/lingbot_video/test_lingbot_video_moe_cuda.py` | Compare the common fused path against an eager LingBot MoE reference on CUDA, including routing and output parity. |
| `benchmarks/lingbot_video/moe_block_benchmark.py` | Add a production-shape block benchmark with latency, memory, backend identification, and optional routing/output artifacts. |
| `benchmarks/lingbot_video/moe_transformer_parity.py` | Adapt official-reference block and transformer comparisons to the common runner; report numerical tolerance checks separately from bitwise equality. |
| `benchmarks/lingbot_video/README.md` | Document benchmark usage, production shapes, routing diagnostics, and numerical acceptance thresholds. |

## Test Plan

**vLLM Version:** `0.28.0`

**vLLM-Omni Commit:**

- Baseline:  `e51fe6ec1`
- Candidate: `0494b518a`

**Environment:** NVIDIA H200 (author-provided designation; runtime reports NVIDIA L20X;
143771 MiB), driver 570.133.20, Python 3.12.13,
PyTorch 2.13.0+cu130, CUDA 13.0, Triton 3.7.1.

**Model:** `robbyant/lingbot-video-moe-30b-a3b`

| Test | Configuration | Repeats |
|---|---|---|
| Production-shape MoE block | 49,145 tokens; hidden 2,048; intermediate 768; 128 experts; top-k 8; 4 groups; top-2 groups | 3 warmups + 20 measured iterations per revision |
| End-to-end serving | 832x480; 121 frames; 40 steps; CFG 3; flow shift 3; seed 42; 24 FPS; concurrency 1 | 1 exact-shape warmup + 3 measured requests per revision |
| Block numerical parity | Identical initialized weights and inputs; compare routing and output tensors | Reuse block artifacts |
| Video quality | Validate metadata, file/decoded hashes, per-frame PSNR/SSIM/MAE, and representative frames | Reuse all 6 measured videos |

Block acceptance thresholds: relative L2 <= 0.5%, cosine similarity >= 0.99999,
and maximum absolute error <= 0.02. No numeric video-quality acceptance threshold
was predefined; video metrics and visual observations are reported separately.

```bash
python -m benchmarks.lingbot_video.moe_block_benchmark --warmups 3 --repeats 20
```

For serving, start each revision separately with the same model and configuration:

```bash
vllm serve robbyant/lingbot-video-moe-30b-a3b --omni \
  --model-class-name LingBotVideoPipeline --port 8099 --log-stats
```

Set `PROMPT` to a fixed test prompt and submit the same request to each revision:

```bash
curl -sS http://localhost:8099/v1/videos \
  -F "prompt=${PROMPT:?Set the same prompt for both revisions}" \
  -F width=832 -F height=480 -F num_frames=121 -F fps=24 \
  -F num_inference_steps=40 -F guidance_scale=3 -F flow_shift=3 -F seed=42
```

Wait for each returned job ID to complete via `GET /v1/videos/{id}` and retrieve
the video via `GET /v1/videos/{id}/content`. Run one warmup and three measured
requests sequentially per revision. Compare decoded videos using PSNR, SSIM, and MAE.

### Performance

| Metric | Baseline | Candidate | Change |
|---|---:|---:|---:|
| MoE block median latency | 25.758 ms | 10.676 ms | **-58.55% (2.413x)** |
| MoE block peak allocated memory | 12,966 MB | 5,508 MB | -57.52% |
| End-to-end median latency | 205.786 s | 175.054 s | **-14.93% (1.176x)** |
| End-to-end p95 latency | 205.990 s | 175.241 s | -14.93% |
| End-to-end peak memory | 81,028 MB | 83,390 MB | **+2,362 MB (+2.92%)** |
| Successful measured requests | 3/3 | 3/3 | No failures |

### Block numerical parity

| Check | Result |
|---|---|
| Finite output | Both revisions pass |
| Gate weights / router logits | Exact match |
| Top-k expert IDs (sorted per token) | Exact match |
| Top-k score maximum absolute error | 0.0009765625 |
| Output relative L2 | 0.3004% — pass |
| Output cosine similarity | 0.99999550 — pass |
| Output maximum absolute error | 0.015625 — pass |
| Output MAE | 0.0005424 |

### Video correctness and quality

All six measured videos are valid H.264, 832x480, 121 frames, and 24 FPS. Each
revision produces identical file and decoded-frame hashes across its three requests.

| Baseline vs. candidate | Result |
|---|---:|
| Mean PSNR | 26.200 dB |
| Mean SSIM | 0.863116 |
| Worst-frame SSIM | 0.831879 (frame 117) |
| Mean MAE (0–255 scale) | 7.441 |
| Worst-frame MAE (0–255 scale) | 13.585 |

### CODE REVIEW

File: vllm_omni/diffusion/models/lingbot_video/pipeline_lingbot_video.py [hunk 1]
```diff
@@ -348,12 +351,24 @@ def __init__(self, *, od_config: OmniDiffusionConfig, prefix: str = ""):
...
+            DiffusersPipelineLoader.ComponentSource(
+                model_or_path=model,
+                subfolder=transformer_subfolder,
+                revision=getattr(od_config, "revision", None),
+                prefix="transformer.",
+                fall_back_to_pt=True,
+            )
+        ]
+        transformer_kwargs = get_transformer_config_kwargs(
+            od_config.tf_model_config,
```
Comment (lishunyang12) [inline vllm_omni/diffusion/models/lingbot_video/pipeline_lingbot_video.py]: Load config from transformer_subfolder too; the selected weights can have a different architecture.

File: vllm_omni/diffusion/models/lingbot_video/pipeline_lingbot_video.py [hunk 2]
```diff
@@ -348,12 +351,24 @@ def __init__(self, *, od_config: OmniDiffusionConfig, prefix: str = ""):
...
+            LingBotVideoTransformer3DModel,
+        )
+        self.transformer = LingBotVideoTransformer3DModel(
+            **transformer_kwargs,
```
Comment (lishunyang12) [inline vllm_omni/diffusion/models/lingbot_video/pipeline_lingbot_video.py:375]: Drop prefix from transformer_kwargs before passing it explicitly; saved native configs already include it.

File: vllm_omni/diffusion/models/lingbot_video/pipeline_lingbot_video.py (hunk 1 above)
Comment (wtz2333) [reply vllm_omni/diffusion/models/lingbot_video/pipeline_lingbot_video.py]: thx,I've fixed it

File: vllm_omni/diffusion/models/lingbot_video/pipeline_lingbot_video.py (hunk 2 above)
Comment (wtz2333) [reply vllm_omni/diffusion/models/lingbot_video/pipeline_lingbot_video.py:375]: OK,I resolved this issue by adding `transformer_kwargs.pop("prefix", None)`
